internal/handlers/common/projection.go

```diff
@@ -233,3 +252,81 @@ func filterFieldArrayElemMatch(k1 string, doc, conditions *types.Document, docVa
 	}
 	return
 }
+
+// filterFieldArraySlice is a function that implements $slice projection query.
+func filterFieldArraySlice(docValue *types.Array, projectionValue any) error {
+	var elementsToSkip, elementsToReturn int
+	switch projectionValue.(type) {
+	// TODO can it have type int64?
```

Comment:
I think PRs like that are where test-driver development really shines. We could write a simple integration test first, note how MongoDB works, adjust the test to pass against MongoDB, then make FerretDB behave the same way.